django/db/migrations/autodetector.py

```diff
@@ -1051,6 +1053,7 @@ def arrange_for_graph(self, changes, graph, migration_name=None):
                     migration.dependencies.append(app_leaf)
                 if i == 0 and not app_leaf:
                     new_name = "0001_%s" % migration_name if migration_name else "0001_initial"
+                    migration.initial = True
```

Comment:
Carl's right -- when I started working on it, I started with the first migration which was easier to find in arrange_for_graphs(), then I realized I need to handle split migrations, which led me to add the code to _detect_changes(), which does cover all initial migrations; I later forgot to remove the code quoted above.